source/agora/node/Ledger.d

```diff
@@ -857,17 +856,6 @@ public class Ledger
         }
 
         assert(validators.length == block.header.validators.count);
-        // Check that more than half have signed
-        auto signed = block.header.validators.setCount;
-        if (signed <= validators.length / 2)
-        {
-            log.error("Block#{}: Signatures are not majority: {}/{}, signers: {}",
-                      block.header.height, signed, validators.length,
-                      block.header.validators);
-            return "The majority of validators hasn't signed this block";
-        }
-
-        log.trace("Checking signature, participants: {}/{}", signed, validators.length);
```

Comment:
I don't think that works, because now `validateBlock` will not check the number of signatures:
https://github.com/bosagora/agora/blob/11ce6e379fd7629e39d31ed80b9e1cc8e02a93cc/source/agora/node/Ledger.d#L746-L798

Which means that we will accept *any* blocks on our endpoints, doesn't it ?

Reply:
I think we might need to be more strict about included signatures when validating blocks we receive. But when we are active in the nomination and externalise when informed by SCP then we can not be strict about % of signatures as we need to add them afterwards. I'll try and implement this.

Reply:
I've added an array to the `ValidatingLedger` to keep track of blocks we are `EXTERNALIZE`ing from `SCP` event. These we can ignore the majority check but for all other cases we want to validate the majority and reject. I think this also covers the case where a `ValidatingLedger` is receiving blocks from the network during catchup after a restart.